submission/proxy.go

```diff
@@ -165,6 +165,10 @@ func (p *Proxy) restartDistributor(ctx context.Context, ll *loglist3.LogList) er
 		// losing ll info. No good.
 		return err
 	}
+	// Don't periodically refresh the roots if we're not going to bother checking them.
+	if d.rootCompatibilityCheckDisabled {
+		return nil
+	}
```

Comment:
Returning `nil` here would make the `restartDistributor` function not working as described. If the root compatibility check is disabled, it should skip the roots refresh instead of the distributor restart.

Reply:
I'll just get rid of this.

`RefreshRoots` already does a no-op if `rootCompatibilityCheckDisabled` is true.